docs/io/ascii/read.rst

```diff
@@ -671,109 +671,163 @@ one class that handles the data, and a reader class that ties it all together.
-  class CommentedHeaderHeader(BasicHeader):
-      """Header class for which the column definition line starts with the
-      comment character.  See the :class:`CommentedHeader` class  for an example.
-      """
-      def process_lines(self, lines):
-          """Return only lines that start with the comment regexp.  For these
-          lines strip out the matching characters."""
-          re_comment = re.compile(self.comment)
-          for line in lines:
-              match = re_comment.match(line)
-              if match:
-                  yield line[match.end():]
-
-      def write(self, lines):
-          lines.append(self.write_comment + self.splitter.join(self.colnames))
-
-
-  class CommentedHeader(Basic):
-      """Read a file where the column names are given in a line that begins with
-      the header comment character. ``header_start`` can be used to specify the
-      line index of column names, and it can be a negative index (for example -1
-      for the last commented line).  The default delimiter is the <space>
-      character.::
-
-        # col1 col2 col3
-        # Comment line
-        1 2 3
-        4 5 6
-      """
-      _format_name = 'commented_header'
-      _description = 'Column names in a commented line'
-
-      header_class = CommentedHeaderHeader
-      data_class = NoHeaderData
-
+  >>> # Note: CommentedHeader is already included in astropy.io.ascii for convenience.
+  >>> class CommentedHeaderHeader(BasicHeader):
+  ...     """Header class for which the column definition line starts with the
+  ...     comment character.  See the :class:`CommentedHeader` class  for an example.
+  ...     """
+  ...     def process_lines(self, lines):
+  ...         """Return only lines that start with the comment regexp.  For these
+  ...         lines strip out the matching characters."""
+  ...         re_comment = re.compile(self.comment)
+  ...         for line in lines:
+  ...             match = re_comment.match(line)
+  ...             if match:
+  ...                 yield line[match.end():]
+  ...
+  ...     def write(self, lines):
+  ...         lines.append(self.write_comment + self.splitter.join(self.colnames))
+  >>>
+  >>>
+  >>> class CommentedHeader(Basic):
+  ...     """Read a file where the column names are given in a line that begins with
+  ...     the header comment character. ``header_start`` can be used to specify the
+  ...     line index of column names, and it can be a negative index (for example -1
+  ...     for the last commented line).  The default delimiter is the <space>
+  ...     character.::
+  ...
+  ...       # col1 col2 col3
+  ...       # Comment line
+  ...       1 2 3
+  ...       4 5 6
+  ...     """
+  ...     _format_name = 'custom_commented_header'
+  ...     _description = 'Column names in a commented line'
+  ...
+  ...     header_class = CommentedHeaderHeader
+  ...     data_class = NoHeaderData
+
+**Application: Write a "fixed_width" table with a "commented_header"**
```

Comment:
You don't want this to be a sub-section?

Reply:
I want this to be on the same level as the existing examples, and they are all just formatted like this with a bold font line.